lib/rack/lint.rb

```diff
@@ -203,6 +206,11 @@ def check_environment(env)
 
         ## <tt>rack.hijack?</tt>:: See below, if present and true, indicates
         ##                         that the server supports partial hijacking.
+        if rack_is_hijack = env[RACK_IS_HIJACK]
+          unless rack_is_hijack == true || rack_is_hijack == false
+            raise LintError, "rack.hijack? must be true or false"
+          end
```

Comment:
This would be a SPEC change, since previously values other than true or false were accepted. Later in SPEC, it states `If +rack.hijack?+ is present in +env+ and truthy` (note `truthy` and not `true`).  So I don't think we should make this change, at least not in this PR.  If you would like to make this SPEC change, please submit a separate PR with SPEC changes and a justification with why you think we should be more strict for `rack.hijack?`.

Reply:
This is my fault, in the original spec for `rack.hijack?` only `true` and `false` were allowed. The word `truthy` was introduced by me: https://github.com/rack/rack/commit/3012643ea6a89fefe8cc0c68d4992531c367c906

Previously it was only `true` or `false`, e.g. https://github.com/rack/rack/blob/f393916baf057bd613290b505150b7ae5da1b9c4/lib/rack/lint.rb#L550

I am okay to do this in a separate PR, but the union of the `if present and true` and `and truthy` only allows for `true` according to the spec, so I don't see this as a breaking change or even a big compatibility issue since this is the same behaviour as Rack 2.x.

Reply:
Let's use a separate PR for this.  As long as the common servers all provide `true`/`false` (and I assume they do for Rack 2 compliance), it should be fine.

Reply:
https://github.com/rack/rack/issues/2294